lnwallet/fee_estimator_test.go

```diff
@@ -89,3 +91,60 @@ func TestStaticFeeEstimator(t *testing.T) {
 		t.Fatalf("expected fee rate %v, got %v", feePerKw, feeRate)
 	}
 }
+
+// TestSparseConfFeeSource checks that SparseConfFeeSource generates URLs and
+// parses API responses as expected.
+func TestSparseConfFeeSource(t *testing.T) {
+	t.Parallel()
+
+	// Test that GenQueryURL returns the URL as is.
+	url := "test"
+	feeSource := lnwallet.SparseConfFeeSource{URL: url}
+	queryURL := feeSource.GenQueryURL()
+	if queryURL != url {
+		t.Fatalf("expected query URL of %v, got %v", url, queryURL)
+	}
+
+	// Test parsing a properly formatted JSON API response.
+	// First, create the response as a bytes.Reader.
+	testFees := map[uint32]uint32{1: 12345, 2: 42, 3: 54321}
```

Comment:
style nit: space out this initialization with some newlines

Reply:
fixed